### DISCUSSION TITLE
Enabling dataview DQL task queries on /search/

### DISCUSSION THREAD

**Discussion by guilherme-kenzo** [coddingtonbear/obsidian-local-rest-api#125] [Ideas]
I'm working on a small project that requires task type dataview DQL queries and realized that obsidian-local-rest-api only supports TABLE type queries. 

Is there any particular reason for this limitation? If not and the project is accepting contributions, I might take a go at it.

**Comment by coddingtonbear** [comment]
Maybe!  It's not immediately obvious to me what the output would look like for a `TASK`-type query, and if you could help me understand what the output would look like maybe I wouldn't mind extending what kinds of dataview DQL we support.

For context, for `TABLE`-type queries, we can just emit JSON objects having keys matching whatever table columns you've selected (see https://github.com/coddingtonbear/obsidian-local-rest-api/blob/433d2fc2f0d445c6b4c5cb8e9599c173605417d3/src/requestHandler.ts#L906-L907).  For many of the other DQL types, not only are there not always obvious things to use as fields, but there might be some hierarchical structure to somehow make use of.

I'm curious to hear what you have in mind, as well as curious to hear what your use case is for these `TASK`-type queries.

**Reply by arthur-er** [reply]
I would like a array of tasks, each being a object with title, status and a metadata object including the other fields

My use case is to be able to export a daily todo list from the vault into other apps

**Reply by coddingtonbear** [reply]
I see; I think you'll also have to find a way of handling the fact that tasks can be hierarchical, too, unfortunately.  What I'd recommend for now is that you implement that as an API extension (see https://github.com/coddingtonbear/obsidian-local-rest-api/wiki/Adding-your-own-API-Routes-via-an-Extension) to see if you can find a way of making that work elegantly.  Let me know if you do, because I'd love to see what you come up with!

**Reply by guilherme-kenzo** [reply]
Great, thanks for the input. I will try to get something working this weekend and will get back to you here.

**Reply by kjarnot-uips** [reply]
> Great, thanks for the input. I will try to get something working this weekend and will get back to you here.

Hi @guilherme-kenzo - did you ever make any progress on this?

**Comment by coddingtonbear** [comment]
I dropped support for DQL queries in v4.0.0; so I think this is unfortunately not going to go any further than iti has.
